File: src/Suframa.Sciex.CrossCutting.DataTransferObject/ViewModel/EnviaRecursoVM.cs
using System.Collections.Generic;

namespace Suframa.Sciex.CrossCutting.DataTransferObject.ViewModel
{
	public class EnviaRecursoVM
	{
		public int? Ano { get; set; }
		public IEnumerable<ArquivoVM> Arquivos { get; set; }
		public string DescricaoServico { get; set; }
		public string DescricaoUnidadeCadastradora { get; set; }
		public IEnumerable<string> Documentos { get; set; }
		public int? IdPapel { get; set; }
		public int? IdProtocolo { get; set; }
		public bool IsConferenciaAdministrativa { get; set; }
		public bool IsIndeferidoAguardandoRecurso { get; set; }
		public string Justificativa { get; set; }
		public string JustificativaIndeferimento { get; set; }
		public IEnumerable<string> MotivoIndeferimento { get; set; }

		public string NumeroProtocolo
		{
			get
			{
				if (NumeroSequencial.HasValue && Ano.HasValue)
				{
					return (NumeroSequencial.Value.ToString("D6") + "/" + Ano.Value).ToString();
				}
				return "";
			}
		}

		public int? NumeroSequencial { get; set; }
	}
}

File: src/Suframa.Sciex.UserInterface/www/app/view-model/EnviaRecursoVM.ts
import { arquivoVM } from './ArquivoVM';

export class enviaRecursoVM {
	ano?: number;
	arquivos: arquivoVM[];
	descricaoServico: string;
	descricaoUnidadeCadastradora: string;
	documentos: string[];
	idProtocolo?: number;
	idPapel?: number;
	isConferenciaAdministrativa: boolean;
	isIndeferidoAguardandoRecurso: boolean;
	justificativa: string;
	justificativaIndeferimento: string;
	motivoIndeferimento: string[];
	numeroProtocolo: string;
	numeroSequencial?: number;
}
